internal/wrappers/cgo/kuzzle/security.go

```diff
@@ -145,221 +131,332 @@ func kuzzle_security_destroy_role(r *C.role) {
-	options := SetQueryOptions(o)
-	res, err := (*kuzzle.Kuzzle)(k.instance).Security.SearchRoles(cToGoSearchFilters(f), options)
-
-	return goToCRoleSearchResult(k, res, err)
+//export kuzzle_security_get_profile_rights
+func kuzzle_security_get_profile_rights(k *C.kuzzle, id *C.char, o *C.query_options) *C.json_result {
+	res, err := (*kuzzle.Kuzzle)(k.instance).Security.GetProfileRights(C.GoString(id), SetQueryOptions(o))
+	return goToCJsonResult(res, err)
 }
 
-//export kuzzle_security_role_delete
-func kuzzle_security_role_delete(r *C.role, o *C.query_options) *C.string_result {
-	result := (*C.string_result)(C.calloc(1, C.sizeof_string_result))
-	opts := SetQueryOptions(o)
-
-	role, err := cToGoRole(r)
-	if err != nil {
-		Set_string_result_error(result, err)
-		return result
-	}
-	res, err := role.Delete(opts)
-
-	return goToCStringResult(&res, err)
+//export kuzzle_security_get_profile_mapping
+func kuzzle_security_get_profile_mapping(k *C.kuzzle, o *C.query_options) *C.json_result {
+	res, err := (*kuzzle.Kuzzle)(k.instance).Security.GetProfileMapping(SetQueryOptions(o))
+	return goToCJsonResult(res, err)
 }
 
-//export kuzzle_security_role_save
-func kuzzle_security_role_save(r *C.role, o *C.query_options) *C.role_result {
+//export kuzzle_security_get_role
+func kuzzle_security_get_role(k *C.kuzzle, id *C.char, o *C.query_options) *C.role_result {
 	result := (*C.role_result)(C.calloc(1, C.sizeof_role_result))
 	options := SetQueryOptions(o)
 
-	role, err := cToGoRole(r)
-	if err != nil {
-		Set_role_result_error(result, err)
-		return result
-	}
-	res, err := role.Save(options)
+	role, err := (*kuzzle.Kuzzle)(k.instance).Security.GetRole(C.GoString(id), options)
 	if err != nil {
 		Set_role_result_error(result, err)
 		return result
 	}
 
-	result.role = goToCRole(r.kuzzle, res, nil)
+	result.role = goToCRole(k, role, nil)
 
 	return result
 }
 
-// --- user
-
-//export kuzzle_security_new_user
-func kuzzle_security_new_user(k *C.kuzzle, id *C.char, d *C.user_data) *C.user {
-	cuser := (*C.user)(C.calloc(1, C.sizeof_user))
+//export kuzzle_security_get_role_mapping
+func kuzzle_security_get_role_mapping(k *C.kuzzle, o *C.query_options) *C.json_result {
+	res, err := (*kuzzle.Kuzzle)(k.instance).Security.GetRoleMapping(SetQueryOptions(o))
+	return goToCJsonResult(res, err)
+}
 
-	cuser.id = id
-	cuser.kuzzle = k
+//export kuzzle_security_get_user
+func kuzzle_security_get_user(k *C.kuzzle, id *C.char, o *C.query_options) *C.user_result {
+	res, err := (*kuzzle.Kuzzle)(k.instance).Security.GetUser(C.GoString(id), SetQueryOptions(o))
+	return goToCUserResult(k, res, err)
+}
 
-	if d != nil {
-		cuser.content = d.content
-		cuser.profile_ids = d.profile_ids
-		cuser.profile_ids_length = d.profile_ids_length
-	}
+//export kuzzle_security_get_user_rights
+func kuzzle_security_get_user_rights(k *C.kuzzle, id *C.char, o *C.query_options) *C.json_result {
```

Comment:
We should not use json_result anymore but string_result as we use char* for json now